---
title: "Castillo-Fernandez et al 2017"
author: "Mary"
date: "8/3/2020"
output: html_document
---





##Loading packages

```{r,packages}
suppressPackageStartupMessages({
    library("R.utils")
    library("missMethyl")
    library("limma")
    library("topconfects")
    library("minfi")
    library("IlluminaHumanMethylation450kmanifest")
    library("MethylToSNP")
    library("RColorBrewer")
    library("IlluminaHumanMethylation450kanno.ilmn12.hg19")
    library("eulerr")
    library("plyr")
    library("gplots")
    library("reshape2")
    library("beeswarm")
  library("BSgenome")
library("MEDIPS")
  library("DESeq2")
library("BSgenome.Hsapiens.UCSC.hg19")
  library("ensembldb")
  library("EnsDb.Hsapiens.v75")
  library("GenomicRanges")
  })
```
```{r,anno}
#Annotation
names(listTables(EnsDb.Hsapiens.v75))
ensgenes<-genes(EnsDb.Hsapiens.v75)
enstranscripts<-transcripts(EnsDb.Hsapiens.v75)
enspromoters<-promoters(EnsDb.Hsapiens.v75,columns=c("gene_id","gene_name"))
head(enspromoters)
head(ensgenes)
```

```{r,functions}
#customised beeswarm chart for MEDIP Seq data
make_beeswarm2<- function(dm,name,mx,groups,n=15) {
    par(mar=c(3,3,1,1))
    NCOLS=5
    NROWS=floor(n/NCOLS)
    if (n %% NCOLS > 0) { NROWS <- NROWS + 1 }
    par(mfrow=c(NROWS, NCOLS))
    topgenes <-  rownames(head(dm[order(dm$pvalue),],n))
    mx<-mx/colSums(mx)*1e6
    ss <- mx[which(rownames(mx) %in% topgenes),]
    n <- 1:n
    g1name=levels(groups)[1]
    g2name=levels(groups)[2]
    g1dat <- ss[n,which(groups == g1name)]
    g2dat <- ss[n,which(groups == g2name)]
    g1l <-lapply(split(g1dat, row.names(g1dat)), unlist)
    g2l <-lapply(split(g2dat, row.names(g2dat)), unlist)
   for (i in n) {
      mydat <- list(g1l[[i]],g2l[[i]])
        beeswarm(mydat,cex=0.4, pch=19,
        las=2, cex.lab=0.6, main=names( g1l )[i] , 
        ylab="",labels = c(g1name,g2name))
      grid()
    }
}

#customised beeswarm chart(confects) for MEDIP Seq data
make_beeswarms_confects2 <- function(confects,name,mx,groups,n=15) {
    par(mar=c(3,3,1,1))
    NCOLS=5
    NROWS=floor(n/NCOLS)
    if (n %% NCOLS > 0) { NROWS <- NROWS + 1 }
    par(mfrow=c(NROWS, NCOLS))
    topgenes <-  head(confects$table,n)$name
    mx<-mx/colSums(mx)*1e6
    ss <- mx[which(rownames(mx) %in% topgenes),]
    n <- 1:n
    g1name=levels(groups)[1]
    g2name=levels(groups)[2]
    g1dat <- ss[n,which(groups == g1name)]
    g2dat <- ss[n,which(groups == g2name)]
    g1l <-lapply(split(g1dat, row.names(g1dat)), unlist)
    g2l <-lapply(split(g2dat, row.names(g2dat)), unlist)

    for (i in n) {
      mydat <- list(g1l[[i]],g2l[[i]])
        beeswarm(mydat,cex=0.4, pch=19,
        las=2, cex.lab=0.6, main=names( g1l )[i] , 
        ylab="",labels = c(g1name,g2name))
      grid()
    }
}



# Annotation 
anno<-function(dge,ensgenes){
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down
list("dge"=dge,"gr_up"=gr_up,"gr_down"=gr_down)
}


# volcano and smear plot
volcanosmear<-function(dge){
sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")

# smear plot optional
plot(log2(dge$baseMean),dge$log2FoldChange,cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 base mean",
ylim=c(-3,3),ylab="log2 fold change"
,pch=19,col="#838383")
points(log2(sig$baseMean),sig$log2FoldChange,cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)
}

#heatmap chart
heatmaps<-function(samplesheet,dge,confects){
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)
rows<-head(confects$table$index,50)
dgem<-as.matrix(dge[rows,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)
}
```



## counts 

```{r,counts c}
counts<-read.table("~/mr.edgeR.c.test.tsv.gz",sep="\t",header=TRUE,row.names=1)
mycol<-max(grep(".bam.counts",colnames(counts)))
counts<-counts[,1:mycol]
rownames(counts)<-paste(counts$chr,counts$start, counts$stop)
counts[,1:4]=NULL
colnames(counts)<-gsub(".bam.counts","",colnames(counts))
```
## metadata
```{r,metadata}
sf1<-read.table("~/castillo_metadata/EGAD00001003159_metadata/delimited_maps/Sample_File.map",stringsAsFactors = FALSE) 
rs1<-read.table("~/castillo_metadata/EGAD00001003159_metadata/delimited_maps/Run_Sample_meta_info.map",sep=";",stringsAsFactors = FALSE)
rs1[,ncol(rs1)]=NULL
md1<-apply(rs1,2,function(x){
  sapply(strsplit(x,"="),"[[",2)
  })
headers<-t(rs1[1,])
colnames(md1)<-sapply(strsplit(headers[,1],"="), "[[",1)
rownames(md1)<-sf1[,1]
md1<-as.data.frame(md1)


sf2<-read.table("~/castillo_metadata/EGAD00001003158_metadata/delimited_maps/Sample_File.map",stringsAsFactors = FALSE) 
rs2<-read.table("~/castillo_metadata/EGAD00001003158_metadata/delimited_maps/Run_Sample_meta_info.map",sep=";",stringsAsFactors = FALSE)
rs2[,ncol(rs2)]=NULL
md2<-apply(rs2,2,function(x){
  sapply(strsplit(x,"="),"[[",2)
  })
headers<-t(rs2[1,])
colnames(md2)<-sapply(strsplit(headers[,1],"="), "[[",1)
rownames(md2)<-sf2[,1]
md2<-as.data.frame(md2)

md <- as.data.frame(rbind(md1,md2))

mdc <- md[grep("C",rownames(md)),]

mdw <- md[grep("W",rownames(md)),]

## CBMC
## number of natural conception 43
nrow(subset(mdc,medical_help_to_conceive=="no"))
## number of ovarian stimulation is 36
nrow(subset(mdc,ovarian_stimulation=="yes"))
## number of gamete intrafallopian transfer is 2
nrow(subset(mdc,gamete_intrafallopian_transfer=="yes"))
## numberof intracytoplasmic sperm injection is 21
nrow(subset(mdc,intracytoplasmic_sperm_injection=="yes"))
## number of ICSI frozen embryo is 9
nrow(subset(mdc,intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
## number of ICSI fresh embryo is 12
nrow(subset(mdc,intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))


## WCB
## number of medical help to conceive is 54
nrow(subset(mdw,medical_help_to_conceive=="no"))
## number of ovarian stimulation is 36
nrow(subset(mdw,ovarian_stimulation=="yes"))
## number of gamete intrafallopian transfer is 2
nrow(subset(mdw,gamete_intrafallopian_transfer=="yes"))
## numberof intracytoplasmic sperm injection is 18
nrow(subset(mdw,intracytoplasmic_sperm_injection=="yes"))
## number of ICSI frozen embryo is 6
nrow(subset(mdw,intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
## number of ICSI fresh embryo is 12
nrow(subset(mdw,intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))
```

based on the numbers, following are the contrasts

CBMC and WMBC

* Natural Vs Ovarian stimulation

* Natural Vs GIFT

* Natural Vs ICSI fresh

* Natural Vs ICSI frozen

* ICSI fresh Vs frozen

* Ovarian stimulation Vs GIFT

* Ovarian stimulation Vs ICSI fresh 

* Ovarian stimulation Vs ICSI frozen

* GIFT Vs ICSI fresh

* GIFT Vs ICSI frozen


## CBMC: Natural Vs Ovarian stimulation
```{r,dm1,fig.width=8,fig.height=8}
NAME = "CBMC_natural_vs_ovarian_stimulation"
samplesheet<-subset(mdc, medical_help_to_conceive=="no" | ovarian_stimulation=="yes")
samplesheet$groups <- factor(samplesheet$ovarian_stimulation,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]
dim(y)
dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

confects <- deseq2_confects(res)
head(confects$table)
volcanosmear(dge)
heatmaps(samplesheet,dge,confects)

#beeswarm charts
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","os",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm1",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm1",mx=y,groups=mygroups,n=15)


myanno<-anno(dge,ensgenes)
head(myanno$dge)
myanno$gr_up
myanno$gr_down
```
## Natural Vs GIFT

```{r,dm2,fig.width=8,fig.height=8}
NAME = "CBMC_natural_vs_gamete_intrafallopian_transfer"
samplesheet<-subset(mdc, medical_help_to_conceive=="no" | gamete_intrafallopian_transfer=="yes")
samplesheet$groups <- factor(samplesheet$gamete_intrafallopian_transfer,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")

#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","gift",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm2",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm2",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## Natural Vs ICSI fresh

```{r,dm3,fig.width=8,fig.height=8}
NAME = "CBMC_natural_vs_intracytoplasmic_sperm_injection_fresh_embryo"
samplesheet<-subset(mdc, medical_help_to_conceive=="no" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]
head(samplesheet)
# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")

#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","icsi",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm3",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm3",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## Natural Vs ICSI frozen

```{r,dm4,fig.width=8,fig.height=8}
NAME = "CBMC_natural_vs_intracytoplasmic_sperm_injection_frozen_embryo"
samplesheet<-subset(mdc, medical_help_to_conceive=="no" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")

#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm4",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm4",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down


```

## ICSI fresh Vs ICSI frozen

```{r,dm5,fig.width=8,fig.height=8}
NAME = "CBMC_intracytoplasmic_sperm_injection_fresh_embryo_vs_intracytoplasmic_sperm_injection_frozen_embryo"
samplesheet<-subset(mdc, intracytoplasmic_sperm_injection=="yes")
samplesheet$groups <- factor(samplesheet$frozen_embryo ,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]


# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","icsi.fh",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm5",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm5",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## ovarian stimulation vs GIFT

```{r,dm6,fig.width=8,fig.height=8}
NAME = "CBMC_ovarian_stimulation_vs_gamete_intrafallopian_transfer"
samplesheet<-subset(mdc, ovarian_stimulation=="yes" | gamete_intrafallopian_transfer=="yes")
samplesheet$groups <- factor(samplesheet$gamete_intrafallopian_transfer,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue",pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
confects <- deseq2_confects(res)
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","os",mygroups)
mygroups<-gsub("yes","gift",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm6",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm6",mx=y,groups=mygroups,n=15)


## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```

## ovarian stimulation vs ICSI fresh

```{r,dm7,fig.width=8,fig.height=8}
NAME = "CBMC_ovarian_stimulation_vs_intracytoplasmic_sperm_injection_fresh_embryo"
samplesheet<-subset(mdc, ovarian_stimulation=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","os",mygroups)
mygroups<-gsub("yes","icsi.fh",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm7",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm7",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## ovarian stimulation vs ICSI frozen

```{r,dm8,fig.width=8,fig.height=8}
NAME = "CBMC_ovarian_stimulation_vs_intracytoplasmic_sperm_injection_frozen_embryo"
samplesheet<-subset(mdc, ovarian_stimulation=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","os",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm8",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm8",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## GIFT vs ICSI fresh

```{r,dm9,fig.width=8,fig.height=8}
NAME = "CBMC_gamete_intrafallopian_transfer_vs_intracytoplasmic_sperm_injection_fresh_embryo"
samplesheet<-subset(mdc, gamete_intrafallopian_transfer=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","gift",mygroups)
mygroups<-gsub("yes","icsi.fh",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm9",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm9",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```

## GIFT vs ICSI frozen

```{r,dm10,fig.width=8,fig.height=8}
NAME = "CBMC_gamete_intrafallopian_transfer_vs_intracytoplasmic_sperm_injection_frozen_embryo"
samplesheet<-subset(mdc, gamete_intrafallopian_transfer=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","gift",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dm10",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dm10",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## WMBC 

```{r,counts w}
counts<-read.table("~/mr.edgeR.w.test.tsv.gz",sep="\t",header=TRUE,row.names=1)
mycol<-max(grep(".bam.counts",colnames(counts)))
counts<-counts[,1:mycol]
rownames(counts)<-paste(counts$chr,counts$start, counts$stop)
counts[,1:4]=NULL
colnames(counts)<-gsub(".bam.counts","",colnames(counts))
```

## Natural Vs Ovarian stimulation

```{r,dmw1,fig.width=8,fig.height=8}
NAME = "WCB_natural_vs_ovarian_stimulation"
samplesheet<-subset(mdw, medical_help_to_conceive=="no" | ovarian_stimulation=="yes")
samplesheet$groups <- factor(samplesheet$ovarian_stimulation,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]
dim(y)
dim(samplesheet)
colnames(y)
rownames(samplesheet)[which(!rownames(samplesheet)%in%colnames(y))]
dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","os",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw1",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw1",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```

## Natural Vs GIFT

```{r,dmw2,fig.width=8,fig.height=8}
NAME = "WBC_natural_vs_gamete_intrafallopian_transfer"
samplesheet<-subset(mdw, medical_help_to_conceive=="no" | gamete_intrafallopian_transfer=="yes")
samplesheet$groups <- factor(samplesheet$gamete_intrafallopian_transfer,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","gift",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw2",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw2",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```

## Natural Vs ICSI fresh

```{r,dmw3,fig.width=8,fig.height=8}
NAME = "WCB_natural_vs_intracytoplasmic_sperm_injection_fresh_embryo"
samplesheet<-subset(mdw, medical_help_to_conceive=="no" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","icsi.fh",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw3",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw3",mx=y,groups=mygroups,n=15)


## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```

## Natural Vs ICSI frozen

```{r,dmw4,fig.width=8,fig.height=8}
NAME = "WCB_natural_vs_intracytoplasmic_sperm_injection_frozen_embryo"
samplesheet<-subset(mdw, medical_help_to_conceive=="no" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","nat",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw4",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw4",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## ICSI fresh Vs ICSI frozen

```{r,dmw5,fig.width=8,fig.height=8}
NAME = "WCB_intracytoplasmic_sperm_injection_fresh_embryo_vs_intracytoplasmic_sperm_injection_frozen_embryo"
samplesheet<-subset(mdw, intracytoplasmic_sperm_injection=="yes")
samplesheet$groups <- factor(samplesheet$frozen_embryo ,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","icsi.fh",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw5",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw5",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## ovarian stimulation vs GIFT

```{r,dmw6,fig.width=8,fig.height=8}
NAME = "WCB_ovarian_stimulation_vs_gamete_intrafallopian_transfer"
samplesheet<-subset(mdw, ovarian_stimulation=="yes" | gamete_intrafallopian_transfer=="yes")
samplesheet$groups <- factor(samplesheet$gamete_intrafallopian_transfer,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","os",mygroups)
mygroups<-gsub("yes","gift",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw6",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw6",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```

## ovarian stimulation vs ICSI fresh

```{r,dmw7,fig.width=8,fig.height=8}
NAME = "WCB_ovarian_stimulation_vs_intracytoplasmic_sperm_injection_fresh_embryo"
samplesheet<-subset(mdw, ovarian_stimulation=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","os",mygroups)
mygroups<-gsub("yes","icsi.fh",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw7",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw7",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## ovarian stimulation vs ICSI frozen

```{r,dmw8,fig.width=8,fig.height=8}
NAME = "WCB_ovarian_stimulation_vs_gamete_intrafallopian_transfer"
samplesheet<-subset(mdw, ovarian_stimulation=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","os",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw8",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw8",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```


## GIFT vs ICSI fresh

```{r,dmw9,fig.width=8,fig.height=8}
NAME = "WCB_gamete_intrafallopian_transfer_vs_intracytoplasmic_sperm_injection_fresh_embryo"
samplesheet<-subset(mdw, gamete_intrafallopian_transfer=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="no"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","gift",mygroups)
mygroups<-gsub("yes","icsi.fh",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw9",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw9",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```

## GIFT vs ICSI frozen

```{r,dmw10,fig.width=8,fig.height=8}
NAME = "WCB_gamete_intrafallopian_transfer_vs_intracytoplasmic_sperm_injection_frozen_embryo"
samplesheet<-subset(mdw, gamete_intrafallopian_transfer=="yes" | (intracytoplasmic_sperm_injection=="yes"&frozen_embryo=="yes"))
samplesheet$groups <- factor(samplesheet$intracytoplasmic_sperm_injection,levels=c("no","yes"))
samplesheet$sex <- factor(samplesheet$gender,levels=c("male","female"))
counts_f<-counts[,colnames(counts) %in% rownames(samplesheet)]

# filter genes with fewer than 10 reads per sample
y <- counts_f[which(rowSums(counts_f)/ncol(counts_f)>=(10)),]
y <- y[,order(colnames(y))]
samplesheet <- samplesheet[order(rownames(samplesheet)),]

dds <- DESeqDataSetFromMatrix(countData = y , colData = samplesheet, design = ~ sex + groups )
res <- DESeq(dds)
z <- results(res)
vsd <- vst(dds, blind=FALSE)
zz <-cbind(as.data.frame(z),assay(vsd))
dge <- as.data.frame(zz[order(zz$pvalue),])
dge[1:10,1:6]

sig <- subset(dge,padj<0.05)
SIG = nrow(sig)
DN = nrow(subset(sig,log2FoldChange<0))
UP = nrow(subset(sig,log2FoldChange>0))
HEADER = paste(NAME, SIG , "DMRs,", UP ,"hypermethylated,", DN, "hypomethylated")


#volcano plot
plot(dge$log2FoldChange,-log10(dge$pvalue),cex=0.6,cex.axis=1.2,cex.lab=1.3, 
xlab="log2 fold change",
ylab="log10 pvalue"
,pch=19,col="#838383")
points(sig$log2FoldChange,-log10(sig$pvalue),cex=0.6,pch=19,col="red")
mtext((HEADER),cex=1.0)

#heatmap chart
mygroups<-as.numeric(samplesheet$groups)
colCols<-gsub("1","yellow",mygroups)
colCols<-gsub("2","orange",colCols)
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 25)
dgem<-as.matrix(dge[1:50,7:ncol(dge)])
heatmap.2(dgem,trace="none",scale="row",margin=c(5,5),col=my_palette,cexRow = 0.5,cexCol = 0.4,ColSideColors =colCols)

#beeswarm charts
confects <- deseq2_confects(res)
mygroups<-samplesheet$groups
mygroups<-gsub("no","gift",mygroups)
mygroups<-gsub("yes","icsi.fz",mygroups)
mygroups<-factor(mygroups)
make_beeswarm2(dm=dge,name="dmw10",mx=y,groups=mygroups,n=15)
make_beeswarms_confects2(confects=confects,name="dmw10",mx=y,groups=mygroups,n=15)

## Annotation
dge$chr<-sapply(strsplit(rownames(dge)," "),"[[",1)
dge$chr<-gsub("chr","",dge$chr)
dge$start<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",2))
dge$end<-as.integer(sapply(strsplit(rownames(dge)," "),"[[",3))

sig<-subset(dge,pvalue<.05)
sig_down<-subset(sig,log2FoldChange<0)
sig_up<-subset(sig,log2FoldChange>0)

gr_up<-GRanges(seqnames=sig_up$chr,
ranges=IRanges(start = sig_up$start , end = sig_up$end)) 
gr_up <- gr_up[which(seqnames(gr_up) %in% seqnames(ensgenes)),]
ol_up<-distanceToNearest(gr_up, ensgenes)
ol_up
gr_up$gene <- ensgenes[subjectHits(ol_up),"symbol"]$symbol
gr_up$distance <- elementMetadata(ol_up)
gr_up

gr_down<-GRanges(seqnames=sig_down$chr,
ranges=IRanges(start = sig_down$start , end = sig_down$end))
gr_down <- gr_down[which(seqnames(gr_down) %in% seqnames(ensgenes)),]
ol_down<-distanceToNearest(gr_down, ensgenes)
ol_down
gr_down$gene <- ensgenes[subjectHits(ol_down),"symbol"]$symbol
gr_down$distance <- elementMetadata(ol_down)
gr_down

```
## summary of DMRs in CBMCs

* Natural Vs Ovarian stimulation
DMR- 1336
hypermethylated- 521
hypomethylated- 815

* Natural Vs GIFT
DMR- 0
hypermethylated- 0
hypomethylated- 0

* Natural Vs ICSI fresh
DMR-338
hypermethylated-304
hypomethylated-34

* Natural Vs ICSI frozen
DMR- 261
hypermethylated- 248
hypomethylated- 13

* ICSI fresh Vs frozen
DMR- 1
hypermethylated- 0
hypomethylated- 0

* Ovarian stimulation Vs GIFT
DMR- 0
hypermethylated- 0
hypomethylated- 0

* Ovarian stimulation Vs ICSI fresh 
DMR- 0
hypermethylated- 0
hypomethylated- 0

* Ovarian stimulation Vs ICSI frozen
DMR- 2
hypermethylated- 0
hypomethylated- 2

* GIFT Vs ICSI fresh
DMR- 0
hypermethylated- 0
hypomethylated- 0

* GIFT Vs ICSI frozen
DMR-2
hypermethylated-2
hypomethylated-0


## summary of DMRs in WBMCs




