---
title: "Hajj 2017 Reanalysis"
author: "The methylation in ART meta-analysis group"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: true
theme: cosmo
---

# Introduction


# Loading packages and functions

These packackes will help us to perform vital steps such as normalisation, filtering, 
differential analysis, etc, and provide information about the array probe annotaions.

These functions provide shortcuts to help with charts and other analysis. They will
eventually be shoved into another Rscript or package but can stay here for now.
```{r,packages}
source("meth_functions.R")
# Annotation
ann450k <- getAnnotation(IlluminaHumanMethylation450kanno.ilmn12.hg19)
myann <- data.frame(ann450k[,c("UCSC_RefGene_Name","Regulatory_Feature_Group","Islands_Name","Relation_to_Island")])
promoters <- grep("Prom",myann$Regulatory_Feature_Group)
```

# Data import
Note: mismatches corrected, one sample removed, see corresponding email from Marcus Dittrich (marcus.dittrich@biozentrum.uni-wuerzburg.de)
 
```{r,data} 

# Downloaded IDAT files from marcus.dittrich@biozentrum.uni-wuerzburg.de 
baseDir <- "Hajj"

# Datasheet converted to CSV, array barcode and location converted to sentrix ID and position, respectively
targets <- read.metharray.sheet(baseDir)
rgSet <- read.metharray.exp(targets = targets)

```


## Normalisation

```{r,norm,fig.cap="Figure 1. Normalisation of bead-array data with SWAN."}
mSet <- preprocessRaw(rgSet)
mSetSw <- SWAN(mSet,verbose=FALSE)
par(mfrow=c(1,2), cex=0.8)
densityByProbeType(mSet[,1], main = "Raw")
densityByProbeType(mSetSw[,1], main = "SWAN")

```

## Filter probes

Here we are running parallel analyses, both including and excluding sex chromosomes.

```{r,filterprobes}
# include sex chromosomes
detP <- detectionP(rgSet)
keep <- rowSums(detP < 0.01) == ncol(rgSet)
mSetSw <- mSetSw[keep,]

# exclude SNP probes
mSetSw <- mapToGenome(mSetSw)
mSetSw_nosnp <- dropLociWithSnps(mSetSw)
mSetSw <- mSetSw_nosnp

# exclude sex chromosomes
keep <- !(featureNames(mSetSw) %in% ann450k$Name[ann450k$chr %in% c("chrX","chrY")])
mSetFlt <- mSetSw[keep,]
```

## Extracting Beta and M-values

```{r,beta_m_vals}
# include sex chromosomes
meth <- getMeth(mSetSw)
unmeth <- getUnmeth(mSetSw)
Mval <- log2((meth + 100)/(unmeth + 100))
beta <- getBeta(mSetSw)

# exclude sex chromosomes
meth <- getMeth(mSetFlt)
unmeth <- getUnmeth(mSetFlt)
Mval_flt <- log2((meth + 100)/(unmeth + 100))
beta_flt <- getBeta(mSetFlt)

```



# MDS analysis

* __MDS by ART__: 

* __MDS by SEX__: .

* __MDS by SEX and ART__:

* __MDS by CHIP ARRAY__:

* __MDS by SLIDE__:



## Scree Plot 
```{r,scree,fig.cap="Figure 2. Screeplot shows contribution of top principal components to the overall variation in the experiment."}
#Scree Plot
par(mfrow=c(2,1))  
myscree(Mval,main="incl sex chr")
myscree(Mval_flt,main="excl sex chr")

```

## MDS plot by ICSI status
```{r,mds1,fig.width = 8 ,fig.height = 8,fig.cap="Figure 3. MDS plot coloured by ICSI status."}
sample_groups <- factor(targets$Group)
colour_palette=brewer.pal(n = length(levels(sample_groups)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$Group))]
plot(1,axes = FALSE,xlab="",ylab="",main="MDS by ICSI status")
legend("center",legend=levels(sample_groups),pch=16,cex=1.2,col=colour_palette)
mydist <- plotMDS(Mval, labels=targets$FileName,col=colours,main="sex chromosomes included")
mydist_flt <- plotMDS(Mval_flt, labels=targets$FileName,col=colours,main="sex chromosomes excluded")
```
## MDS plot by sex
```{r,mds2,fig.width = 8 ,fig.height = 8, fig.cap="Figure 4. MDS plot coloured by sex."}
sample_groups <- factor(targets$Gender)
colour_palette=brewer.pal(n = length(levels(sample_groups)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$Gender))]
plot(1,axes = FALSE,xlab="",ylab="",main="MDS by sex")
legend("center",legend=levels(sample_groups),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$FileName,col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$FileName,col=colours,main="sex chromosomes excluded")
```


## MDS by ART and sex
```{r,mds4,fig.width = 8 ,fig.height = 8, fig.cap="Figure 5. MDS plot coloured by ART classification and sex."}
targets$ARTgender <- paste(targets$Group, targets$Gender, sep=("_"))
sample_groups <- factor(targets$ARTgender)
colour_palette=brewer.pal(n = length(levels(sample_groups)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$ARTgender))]
plot(1,axes = FALSE,xlab="",ylab="",main="MDS by ART and sex")
legend("center",legend=levels(sample_groups),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$FileName, col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$FileName, col=colours,main="sex chromosomes excluded")
```
## MDS by chip array
```{r,mds5,fig.width = 8 ,fig.height = 8, fig.cap="Figure 6. MDS plot coloured by chip array"}

sample_groups <- factor(targets$Array)

colour_palette=brewer.pal(n = length(levels(sample_groups)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$Array))]
plot(1,axes = FALSE,xlab="",ylab="",main="MDS by chip")
legend("center",legend=levels(sample_groups),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$FileName, col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$FileName, col=colours,main="sex chromosomes excluded")

```

## MDS by Slide
```{r,mds6,fig.width = 8 ,fig.height = 8, fig.cap="Figure 7. MDS plot coloured by slide"}
sample_groups <- factor(targets$Slide)

colour_palette=brewer.pal(n = length(levels(sample_groups)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$Slide))]
plot(1,axes = FALSE,xlab="",ylab="",main="MDS by slide")
legend("center",legend=levels(sample_groups),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$FileName, col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$FileName, col=colours,main="sex chromosomes excluded")

```

## MDS by Maternal Age
```{r,mds7,fig.width = 8 ,fig.height = 8, fig.cap="Figure 8. MDS plot coloured maternal age"}
targets$catAge <- cut(targets$AgeMother, breaks = c(0,27,35,44), labels = c("A","B","C")) 

sample_groups <- factor(targets$catAge)

colour_palette=brewer.pal(n = length(levels(sample_groups)), name = "Paired")
colours <- colour_palette[as.integer(factor(targets$catAge))]
plot(1,axes = FALSE,xlab="",ylab="",main="MDS by Maternal Age")
legend("center",legend=levels(sample_groups),pch=16,cex=1.2,col=colour_palette)
plotMDS(mydist, labels=targets$FileName, col=colours,main="sex chromosomes included")
plotMDS(mydist_flt, labels=targets$FileName, col=colours,main="sex chromosomes excluded")

```



## Differential analysis

* __NAT (46) vs ICSI (47)__: 



The differential analysis is centred around limma to identify differentially methylated probes. 
TopConfects was also run to obtain the probes with the largest confident effect (topconfect).
There are five outputs below:

1. Volcano plot (limma result).

2. Beeswarm plot (top probes by limma p-value).

3. Heatmap (top probes by limma p-value).

4. Beeswarm plot (top probes by topconfect ranking).

5. Heatmap (top probes by topconfect ranking). (TODO)

### 1

```{r,nat_vs_icsi_incl, fig.width = 8 ,fig.height = 8}
# include sex chromosomes

samplesheet <- targets
samplesheet$Basename <- sapply(strsplit(samplesheet$Basename, "/"), "[[", 3)

groups <- factor(samplesheet$Group,levels=c("Control","ICSI"))
sex <- factor(samplesheet$Gender,levels=c("male","female"))

    
top_nat_vs_icsi_inc <- dm_analysis(samplesheet=samplesheet,
    sex=sex,groups=groups,mx=Mval,name="top_nat_vs_icsi_inc",
    myann=myann ,beta= beta)

mx=Mval
name="top_nat_vs_icsi_inc"

head(top_nat_vs_icsi_inc$dma)
head(top_nat_vs_icsi_inc$dmr)
```

```{r,nat_vs_icsi_exc, fig.width = 8 ,fig.height = 8 }
# exclude sex chromosomes

samplesheet <- targets
samplesheet$Basename <- sapply(strsplit(samplesheet$Basename, "/"), "[[", 3)

groups <- factor(samplesheet$Group,levels=c("Control","ICSI"))
sex <- factor(samplesheet$Gender,levels=c("male","female"))

    
top_nat_vs_icsi_exc <- dm_analysis(samplesheet=samplesheet,
    sex=sex,groups=groups,mx=Mval_flt,name="top_nat_vs_icsi_exc",
    myann=myann ,beta= beta)

# Allele
top_nat_vs_icsi_exc$dma$unmeth <- "T"
top_nat_vs_icsi_exc$dma$meth <- "C"
top_nat_vs_icsi_exc$fit$SE <- sqrt(top_nat_vs_icsi_exc$fit$s2.post) * top_nat_vs_icsi_exc$fit$stdev.unscaled


# Extract required columns from dma
top_nat_vs_icsi_metal <-top_nat_vs_icsi_exc$dma[,c("Row.names", "meth", "unmeth", "AveExpr", "P.Value")]
head(top_nat_vs_icsi_metal)

# Convert fit outputs to dataframes
fitCE <- as.data.frame(top_nat_vs_icsi_exc$fit$coefficients)
fitCE$Row.names <- row.names(fitCE)
fitCE <- fitCE[,c("Row.names", "groupsICSI")]
names(fitCE)[2]<- "coefficient"


fitSE <- as.data.frame(top_nat_vs_icsi_exc$fit$SE)
fitSE$Row.names <- row.names(fitSE)
fitSE <- fitSE[,c("Row.names", "groupsICSI")]
names(fitSE)[2]<- "SE"

# Merge Datasets
top_nat_vs_icsi_metal <- merge(top_nat_vs_icsi_metal, fitCE)
top_nat_vs_icsi_metal <- merge(top_nat_vs_icsi_metal, fitSE)

# Number of effective participants
controls <- samplesheet[samplesheet$Group == "Control",]
cases <- samplesheet[samplesheet$Group == "ICSI",]
ctrl <- nrow(controls)
cses <- nrow(cases)
Neff <- 4/((1/cses)+(1/ctrl))

top_nat_vs_icsi_metal$N <- Neff
top_nat_vs_icsi_metal
# Output for Meta-analysis
write.table(top_nat_vs_icsi_metal, file="NATvsICSI/hajj_top_nat_vs_icsi_metal.tsv",sep="\t",quote=FALSE, row.names = FALSE) 


head(top_nat_vs_icsi_exc$dma)
head(top_nat_vs_icsi_exc$dmr) 
head(top_nat_vs_icsi_metal)
```
```

## Enrichment analysis

We will be using the recently published package mitch to perform enrichment analysis, using
average promoter methylation change as an indicaator of gene activity.
Enrichment will be tested with the mitch package.

```{r,genesets}
library("mitch")
# gene sets
download.file("https://reactome.org/download/current/ReactomePathways.gmt.zip", 
    destfile="ReactomePathways.gmt.zip")
unzip("ReactomePathways.gmt.zip",overwrite = TRUE)
genesets <- gmt_import("ReactomePathways.gmt")

```

# Run 1D enrichment analysis

One dimensional enrichment analysis with REACTOME gene sets using average promoter methylation t-statistic.

```{r,1danalysis}
top_nat_vs_icsi_exc_mitch <- run_mitch_1d(dma= top_nat_vs_icsi_exc$dma, name="top_nat_vs_icsi_exc_mitch")
head(top_nat_vs_icsi_exc_mitch,50)

```

## Todo


## References
